code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -374,6 +374,17 @@
 		return
 	if(default_deconstruction_crowbar(I))
 		return
+	if(panel_open)
+		if(istype(I, /obj/item/stock_parts/cell))
+			I.forceMove(src) // Force it out of our hands so we can put the old cell in it
+			if(!user.put_in_hands(cell))
+				cell.forceMove(get_turf(src))
```

Comment:
This line shouldn't be necessary, `put_in_hands` moves to drop location if it fails.